### PR TITLE
feat(TCOMP-1761): Support of complete schema definition in Studio

### PR DESCRIPTION
**Pull request by undx**
**Support of complete schema definition in Studio**
https://jira.talendforge.org/browse/TCOMP-1761

**Related PR**
- https://github.com/Talend/component-runtime/pull/446
- https://github.com/Talend/tdi-studio-se/pull/5270

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/configuration.javajet [hunk 1]
```diff
@@ -73,54 +78,85 @@ final java.util.Map<String, String> registry_metadata_<%=cid%> = new java.util.H
...
                         <%
                         }
                     }
                     if (!hasDynamic){
-                        for(int i = 0; i < columns.size(); i++) {
-                            IMetadataColumn column = columns.get(i);
-                        %>
-                        configuration_<%=cid%>.put("<%=parameterName%>[<%=i%>]", "<%=column.getLabel()%>");
-                        <%
+                        String [] a = java.util.Optional.ofNullable(p.getListItemsDisplayCodeName()).orElse(new String[0]);
```
Comment (ypiel-talend) [inline main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/configuration.javajet:88]: 'a' is not a good variable name.

**Comment by ypiel-talend** [comment]
Is there any new unit test with List<StudioMetaData> ?

**Review by ypiel-talend — CHANGES_REQUESTED** [review]
simple remarks

File: main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/configuration.javajet (hunk 1 above)
Comment (undx) [reply main/plugins/org.talend.sdk.component.studio-integration/jet_stub/generic/configuration.javajet:88]: sure!

**Review by ypiel-talend — APPROVED** [review]
LGTM

**Pull request merged**